### ISSUE TITLE
snippy-core, variants in .full.aln lost in .vcf

### ISSUE THREAD

**Issue by MartinaTamb** [tseemann/snippy#330]
Hello,
I observed that some variants present in the .full.aln obtained from snippy-core are not present in the .vcf. 
Those variants are present both in the samplename/snps.vcf and samplename/snps.filt.vcf so they should have passed the quality check.
 I also noticed that in the .full.aln, for some isolates, the variants not found in the .vcf from snippy-core are written in uppercase instead of lowercase. What does it mean?
Thank you very much for your help!

**Comment by tseemann** [comment]
1. snippy-core is only SNPs, not INS or DEL. is that what you mean?

2. If there is no difference to the reference, then there is no "variant' in a VCF, so i patch the reference allele into the alignment.

3. Read https://github.com/tseemann/snippy#why-is-corefullaln-an-alphabet-soup

**Comment by actledge** [comment]
> 1. snippy-core is only SNPs, not INS or DEL. is that what you mean?
> 2. If there is no difference to the reference, then there is no "variant' in a VCF, so i patch the reference allele into the alignment.
> 3. Read https://github.com/tseemann/snippy#why-is-corefullaln-an-alphabet-soup

I have met the same problem,
and I found that, a plenty of SNPs(the type column in snps.vcf is "snp" or "complex") does exists in Snippy output of individual genome, and as above mentioned, they should have passed the quailty check. But the were not exists in snippy-core output,  and this occur often by large fragment of reference. For example, all snps located before the position 2540 of reference were not included in core.vcf:
here I copyed some snps from snps.vcf of a sample:
8-911   52  .   CT  GG  6319.61 .   AB=0;AO=232;DP=232;QA=8529;QR=0;RO=0;TYPE=mnp   GT:DP:RO:QR:AO:QA:GL    1/1:232:0:0:232:8529:-767.536,-69.839,0                                                                                             
8-911   60  .   G   C   6786.77 .   AB=0;AO=225;DP=225;QA=8216;QR=0;RO=0;TYPE=snp   GT:DP:RO:QR:AO:QA:GL    1/1:225:0:0:225:8216:-739.382,-67.7318,0                                                                                            
8-911   67  .   AG  A   5667.43 .   AB=0;AO=218;DP=219;QA=7934;QR=0;RO=0;TYPE=del   GT:DP:RO:QR:AO:QA:GL    1/1:219:0:0:218:7934:-713.767,-65.6245,0                                                                                            
8-911   2201    .   C   G   3976.54 .   AB=0;AO=160;DP=162;QA=5862;QR=0;RO=0;TYPE=snp   GT:DP:RO:QR:AO:QA:GL    1/1:162:0:0:160:5862:-527.308,-48.1648,0                                                                                        
8-911   2216    .   T   C   32570.8 .   AB=0;AO=1302;DP=1302;QA=47873;QR=0;RO=0;TYPE=snp    GT:DP:RO:QR:AO:QA:GL    1/1:1302:0:0:1302:47873:-4306.44,-391.941,0                                                                                 
8-911   2225    .   TCG CCT 40421.4 .   AB=0;AO=1614;DP=1615;QA=59118;QR=0;RO=0;TYPE=complex    GT:DP:RO:QR:AO:QA:GL    1/1:1615:0:0:1614:59118:-5317.6,-485.862,0                                                                             
8-911   2237    .   CAGG    TAGA    52621.1 .   AB=0;AO=2107;DP=2114;QA=76844;QR=0;RO=0;TYPE=complex    GT:DP:RO:QR:AO:QA:GL    1/1:2114:0:0:2107:76844:-6912.2,-634.27,0                                                                                                                              
....
.... (after here, the snps included in core.vcf)
8-911   2540    .   TGATTG  CGACTC  230134  .   AB=0;AO=7027;DP=7058;QA=256264;QR=0;RO=0;TYPE=complex   GT:DP:RO:QR:AO:QA:GL    1/1:7058:0:0:7027:256264:-23051.1,-2115.34,0                                                                    
8-911   2600    .   C   T   141227  .   AB=0;AO=4581;DP=4596;QA=165216;QR=296;RO=8;TYPE=snp GT:DP:RO:QR:AO:QA:GL    1/1:4596:8:296:4581:165216:-14834.6,-1354.72,0                                                                              
8-911   2609    .   C   G   126308  .   AB=0;AO=4049;DP=4052;QA=146861;QR=25;RO=1;TYPE=snp  GT:DP:RO:QR:AO:QA:GL    1/1:4052:1:25:4049:146861:-13207.9,-1216.94,0                                                                               
8-911   2615    .   A   G   115443  .   AB=0;AO=3668;DP=3668;QA=133747;QR=0;RO=0;TYPE=snp   GT:DP:RO:QR:AO:QA:GL    1/1:3668:0:0:3668:133747:-12030.7,-1104.18,0                                                                                
8-911   2621    .   T   G   103267  .   AB=0;AO=3264;DP=3264;QA=119159;QR=0;RO=0;TYPE=snp   GT:DP:RO:QR:AO:QA:GL    1/1:3264:0:0:3264:119159:-10718.6,-982.562,0                                                                                
8-911   2643    .   TTG CTA 22990.1 .   AB=0;AO=725;DP=730;QA=25959;QR=0;RO=0;TYPE=complex  GT:DP:RO:QR:AO:QA:GL    1/1:730:0:0:725:25959:-2335.17,-218.247,0                                                                                   
8-911   2651    .   TTC GGG 14544.7 .   AB=0;AO=495;DP=495;QA=18045;QR=0;RO=0;TYPE=complex  GT:DP:RO:QR:AO:QA:GL    1/1:495:0:0:495:18045:-1623.51,-149.01,0 
                                                                                                                                                 
I copyed the start point of the core.vcf, it start from 2540:
8-911   2540    .   T   C   .   PASS    TYPE=snp    GT  1   0   1   0   1   1   1   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0                                                                                           8-911   2543    .   T   C   .   PASS    TYPE=snp    GT  1   1   1   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0                                                                                           8-911   2545    .   G   C,T .   PASS    TYPE=snp    GT  1   0   1   0   2   2   2   1   0   0   0   0   0   1   0   0   0   0   1   0   0   0   0   0                                                                                           8-911   2549    .   C   T   .   PASS    TYPE=snp    GT  0   0   0   0   1   1   1   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0                                                                                           8-911   2561    .   A   G   .   PASS    TYPE=snp    GT  0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0   0   0   1   1                                                                                           8-911   2569    .   T   C   .   PASS    TYPE=snp    GT  0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0                                                                                           8-911   2580    .   C   T   .   PASS    TYPE=snp    GT  0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0   0                                                                                           8-911   2594    .   G   A   .   PASS    TYPE=snp    GT  0   0   0   0   1   1   1   1   0   0   1   0   0   0   0   0   0   1   1   0   0   0   0   0                                                                                           8-911   2600    .   C   T   .   PASS    TYPE=snp    GT  1   0   0   0   0   0   0   0   0   0   0   0   0   1   0   0   0   0   0   0   0   0   0   0                                                                                           8-911   2609    .   C   A,G,T   .   PASS    TYPE=snp    GT  2   0   0   1   2   2   2   3   0   0   2   0   1   2   0   0   0   2   1   0   0   0   0   0

So...  I can't figured out, why a whole fragment snps are ignored, even if they exist in the snps.vcf of a single sample.

And this also occured at other locations in the core-vcf as well.



Thank you very much.
